### PR TITLE
Aligning array and deque api: Add deque::blit_to

### PR DESCRIPTION
**Pull request by ethereal-dream**

### CODE REVIEW

**Review by Lampese — COMMENTED** [review]
Please add more test for edge cases to keep it right.

File: deque/deque.mbt [hunk 1]
```diff
@@ -1759,3 +1759,51 @@ pub fn[A] shuffle(self : Deque[A], rand~ : (Int) -> Int) -> Deque[A] {
...
+    let new_dst = UninitializedArray::make(dst_offset + len)
+    UninitializedArray::unsafe_blit(new_dst, 0, dst.buf, 0, dst.length())
+    UninitializedArray::unsafe_blit(new_dst, dst_offset, self.buf, src_offset, len)
+    dst.buf = new_dst
+    dst.len = dst_offset + len
+    dst.tail = dst.len - 1
+  } else {
+    UninitializedArray::unsafe_blit(dst.buf, dst_offset, self.buf, src_offset, len)
+  }
+}
```
Comment (Lampese) [inline deque/deque.mbt]: No end line? Did this code formatted?

File: deque/deque.mbt [hunk 2]
```diff
@@ -1763,3 +1763,63 @@ pub fn[A] shuffle(self : Deque[A], rand~ : (Int) -> Int) -> Deque[A] {
...
+    dst.tail = dst.len - 1
+  } else {
+    UninitializedArray::unsafe_blit(
+      dst.buf,
+      dst_offset,
+      self.buf,
+      src_offset,
+      len,
+    )
+  }
```
Comment (FrozenLemonTee) [inline deque/deque.mbt:1824]: @Guest0x0 @Lampese 
Consider this case:
```moonbit
test "blit_to_wrap_around" {
  let src = @deque.of([1, 2, 3, 4, 5])
  ignore(src.pop_front())
  src.push_back(6)
  assert_eq(src.to_array(), [2, 3, 4, 5, 6])
  inspect(src.as_views(), content="([2, 3, 4, 5], [6])")
  let dst = @deque.of([0, 0, 0, 0, 0])
  @deque.Deque::blit_to(src, dst, len=5, src_offset=0, dst_offset=0)
  assert_eq(dst.to_array(), [2, 3, 4, 5, 6])
  inspect(dst.as_views(), content="([2, 3, 4, 5, 6], [])")
}
```

Comment (FrozenLemonTee) [reply deque/deque.mbt:1824]: This function can not work properly because it assumes `self` has continuous layout of the elements

**Comment by Lampese** [comment]
Close because
- #2709
